### PR TITLE
User quests

### LINKED ISSUE #13: Endpoint - GET Current Quests for User
- [ ] Create endpoint to GET Current Quests for User 

**Request**
```
GET api/v1/users/:id/quests?completion_status=false
```

**Response**
``` 
status: 200
body: 

{
  "data":  {
      "id": "Null",
      "type": "quests",
      "attributes": { 
            "quests": [ 
                 { "id": "1",
                   "type": "active",
                   "name": "Slay the Wildabeast", 
                   "xp": "200", 
                   "encounter_req": "3",
                   "level": "1",
                   "progress": "1"
                  }, 
                 { "id": "2",
                   "type": "passive", 
                   "name": "Brew a Potion of Confidence", 
                   "xp": "200", 
                   "encounter_req": "3"
                   "level": "1",
                   "progress": "2"
                  }, 
                  { "id": "3",
                   "type": "supportive",
                   "name": "Mend A Wizard's Spellbook", 
                   "xp": "200", 
                   "encounter_req": "3"
                   "level": "1",
                   "progress": "1"
                  }, 
               ]
            }
        }
    }
}
```

- [ ] Endpoint returns all quests associated with a specific user (`user_quests`) where `completion_status` equals `false`

### LINKED ISSUE #15: Endpoint - GET Completed Quests for User
- [ ] Endpoint to return all quests for a user where `completion_status` is `true` 

```
GET api/v1/users/:id/quests?completion_status=true 
```

**Response**
``` 
status: 200
body: 

{
  "data":  {
      "id": "Null"
      "type": "quests",
      "attributes": { 
            "quests": [ 
                 { "id": "1",
                   "type": "active",
                   "name": "Slay the Wildabeast", 
                   "xp": "200", 
                   "encounter_req": "3",
                   "progress": "1" 
                  }, 
                 { "id": "2",
                   "type": "passive", 
                   "name": "Brew a Potion of Confidence", 
                   "xp": "200", 
                   "encounter_req": "3",
                   "progress": "1" 
                  }, 
                  { "id": "3",
                   "type": "supportive",
                   "name": "Mend A Wizard's Spellbook", 
                   "xp": "200", 
                   "encounter_req": "3",
                   "progress": "1"
                  }, 
                  { "id": "4",
                   "type": "passive", 
                   "name": "Unlock the Tome of Job Readiness", 
                   "xp": "200", 
                   "encounter_req": "3",
                   "progress": "1"
                  }, 
                  { "id": "5",
                   "type": "supportive",
                   "name": "Write A Sonnet For Inspiration", 
                   "xp": "200", 
                   "encounter_req": "3",
                   "progress": "1"
                  }, 
               ]
            }
        }
    }
}
```

### PR DESCRIPTION
**Pull request by c-bartell**
### Description

Adds the `users/:id/quests` endpoint, allowing the front end to fetch a user's quest data filtered by completion status. The format of the endpoint and request have been updated and can be seen below:

Baller screenshot of the endpoint in postman

### Closes issue(s)
Closes #13  
Closes #15 
### Testing
Tests were added for happy and sad path, and are passing.
### Changes include
- [ ] Bugfix (non-breaking change that solves an issue)
- [X] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (change that is not backwards-compatible and/or changes current functionality)

### Checklist
- [X] I have written tests for this code (happy & sad)
- [ ] I have updated the Readme
- [X] I ran full test suite - all tests passing

### Other comments
YEET

### CODE REVIEW

File: api/resources/user_quests.py [hunk 1]
```diff
@@ -0,0 +1,72 @@
+import datetime
+import json
+from flask import jsonify
+
+import bleach
+# import jsons
```
Comment (smcunning) [inline api/resources/user_quests.py]: Could we clean up the unneeded commented out code in this file?

File: api/database/models.py [hunk 2]
```diff
@@ -112,7 +112,6 @@ def __init__(self, name, xp, encounter_req, type, level):
         self.level = level
 
 
-
```
Comment (jakeheft) [inline api/database/models.py:115]: Some of these classes are double spaced and some are single spaced, let's come to an agreement on newline conventions

File: api/resources/user_quests.py (hunk 1 above)
Comment (jakeheft) [reply api/resources/user_quests.py]: ^^^

**Review by GeorgieGirl24 — APPROVED** [review]

**Review by jakeheft — CHANGES_REQUESTED** [review]
Nice work on the test! Just needs comment and spacing addressed

**Review by jakeheft — APPROVED** [review]
Thanks for making the changes Shaunda, looks great

**Pull request merged**